dvc/commands/experiments/__init__.py

```diff
@@ -58,3 +58,42 @@ def add_parser(subparsers, parent_parser):
     for cmd in SUB_COMMANDS:
         cmd.add_parser(experiments_subparsers, parent_parser)
     fix_plumbing_subparsers(experiments_subparsers)
+
+
+def add_rev_selection_flags(
+    experiments_subcmd_parser, command: str, default: bool = True
+):
+    experiments_subcmd_parser.add_argument(
+        "-A",
+        "--all-commits",
+        action="store_true",
+        default=False,
+        help=(
+            f"{command} all experiments in the repository "
+            "(overrides `--rev` and `--num`)."
+        ),
+    )
+    default_msg = " (HEAD by default)" if default else ""
+    msg = (
+        f"{command} experiments derived from the specified `<commit>` as "
+        f"baseline{default_msg}."
+    )
+    experiments_subcmd_parser.add_argument(
+        "--rev",
+        type=str,
+        default=None,
+        help=msg,
+        metavar="<commit>",
+    )
+    experiments_subcmd_parser.add_argument(
+        "-n",
+        "--num",
+        type=int,
+        default=1,
+        dest="num",
+        metavar="<num>",
+        help=(
+            f"{command} experiments from the `--rev` baseline and from `num` "
+            "commits before it. Give `-1` to include all the parent commits."
```

Comment:
From the long discussion in https://github.com/iterative/dvc.org/pull/3191#pullrequestreview-855848503 it seems that the behavior around `n=1` has an issue: it should show/list experiments from the given `--rev <commit>` (`HEAD` by default) AND from it's parent commit (the `1` value). Otherwise there's no way to get list/show that configuration.

The default value should be `n=0` for the current default behavior. Thanks

Reply:
Maybe we should follow the `git` behavior? 

https://github.com/iterative/dvc/pull/7245#discussion_r797801555

Reply:
Thanks, @karajan1001. @jorgeorpinel, please ping if you still have questions/concerns. Otherwise, I think we should move forward with the git behavior.

Reply:
> git log -n 0 returns nothing, while git log -n -1 and git log -n -2 return all history. If we want consistency with git

Agreed.

I'd also insist that the metavar name can be improved, at least to `<number>` (consistent with [git log](https://git-scm.com/docs/git-log#_commit_limiting)).